packages/schemas/src/types/saml-application.ts

```diff
@@ -0,0 +1,43 @@
+import { type z } from 'zod';
+
+import { Applications } from '../db-entries/application.js';
+import { SamlApplicationConfigs } from '../db-entries/saml-application-config.js';
+import { SamlApplicationSecrets } from '../db-entries/saml-application-secret.js';
+
+import { applicationCreateGuard } from './application.js';
+
+const samlAppConfigGuard = SamlApplicationConfigs.guard.pick({
+  attributeMapping: true,
+  entityId: true,
+  acsUrl: true,
+});
+
+export const samlApplicationCreateGuard = applicationCreateGuard
+  .pick({
+    name: true,
+    description: true,
+    customData: true,
+  })
+  .extend({
+    // The reason for encapsulating attributeMapping and spMetadata into an object within the config field is that you cannot provide only one of `attributeMapping` or `spMetadata`. Due to the structure of the `saml_application_configs` table, both must be not null.
+    config: samlAppConfigGuard.partial().optional(),
+  });
+
+export type CreateSamlApplication = z.infer<typeof samlApplicationCreateGuard>;
+
+export const samlApplicationResponseGuard = Applications.guard
+  .merge(
+    // Partial to allow the optional fields to be omitted in the response.
+    // When starting to create a SAML application, SAML configuration is optional, which can lead to the absence of SAML configuration.
+    samlAppConfigGuard
+  )
+  .extend({
+    secrets: SamlApplicationSecrets.guard
+      .omit({
+        tenantId: true,
+        applicationId: true,
+      })
+      .array(),
```

Comment:
How about defining a specific API for the secret? I think this is how our application endpoints were designed.

Reply:
We can, this type guard definition can be refactored when we add `POST /saml-applications/:id/secret` API.

Reply:
If that is the case, can we remove the secret definitions here? Seems not useful at the moment.

Reply:
I tend to keep it for now, even if we remove this part now, the subsequent PR will need to add it back. Keeping it can also save the effort of repeated rebasing.

Reply:
Also, in [PR#6832](https://github.com/logto-io/logto/pull/6832/files), we also need to hide `privateKey` from secrets.

Reply:
You can regard this as a placeholder since it is subject to change soon.

Reply:
Discussed offline, do not return `secrets` in SAML application related APIs and will use a separated `GET /saml-applications/:id/secrets` API.